.github/workflows/deploy-doc-site.yml

```diff
@@ -0,0 +1,82 @@
+name: Deploy documentation site
+
+on:
+  push:
+    tags: [ v* ]
+  pull_request:
+    # Run this workflow as a dry run when it's edited, so we can test it.
+    paths: .github/workflows/deploy-doc-site.yml
+
+jobs:
+  deploy:
+    runs-on: macOS-latest
+    env:
+      # Default to deploying the current REF. If this is a pull request run, that ref will be invalid
+      # (local-only merge commit), so we'll override it with the GITHUB_HEAD_REF below.
+      DEPLOY_REF: ${{ github.ref }}
+    steps:
+      - uses: actions/checkout@v2
+
+      # Gradle caches (keys must match those defined in kotlin.yml)
+      # Don't use the gradle wrapper cache, since there's only one job we're downloading the whole wrapper once either way.
+      - name: Cache gradle artifacts
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches
+          key: gradle-${{ hashFiles('**/gradle/wrapper/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle*') }}-${{ github.sha }}
+          restore-keys: |
+            gradle-${{ hashFiles('**/gradle/wrapper/gradle-wrapper.properties') }}-${{ hashFiles('**/*.gradle*') }}-
+
+      # Swift caches (keys must match those defined in swift.yml)
+      - name: Load gem cache
+        uses: actions/cache@v1
+        with:
+          path: .bundle
+          key: gems-${{ hashFiles('Gemfile.lock') }}
+
+      - name: Set up Swift environment
+        run: |
+          # Set global bundle path so it gets used by build_swift_docs.sh running in the nested repo as well.
+          bundle config --global path "$(pwd)/.bundle"
+          bundle check || bundle install
+          # Don't need to run pod gen, the website script does that itself.
+          brew install sourcedocs
+          sudo xcode-select -s /Applications/Xcode_11.3.1.app
```

Comment:
nit: We should probably move the xcodeversion to some global config, but we can clean it up later.